index.html

```diff
@@ -6281,7 +6336,43 @@ <h3><dfn>Execute Async Script</dfn></h3>
       and data <var>result</var>.
 </ol>
 </section> <!-- /Execute Async Script -->
-</section> <!-- /Executing Script -->
+<section>
+ <h3><dfn>Pin Script</dfn></h3>
+
+ <table class="simple jsoncommand">
+  <tr>
+   <th>HTTP Method</th>
+   <th>URI Template</th>
+  </tr>
+  <tr>
+   <td>POST</td>
+   <td>/session/{<var>session id</var>}/execute/pin/{<var>name</var>}</td>
```

Comment:
Is it just me, or is it a bit weird to pick one piece of data from the query parameters and the other from the request body?

Reply:
I was modeling this after getting attribute value. We do that from the endpoint and not the request body.

Reply:
Getting an element attribute is conceptually different because it is idempotent. In this case you are modifying state, and the _name_ field in the URL template is a disguised attempt simulating query parameters (`?foo=bar`) in a POST request.

In RESTful APIs the model is usually that the POST request returns an identifier you can later use in a GET request. If we hypothetically imagine we wanted a getter for the pinned script, `GET /session/{session id}/execute/pin/{name}` would be acceptable.

I also have two other concerns:

1. Script names that include “special” characters that are not permitted in URLs must be encoded, for example `foo bar baz æøå` becomes `foo%20bar%20baz%20%C3%A6%C3%B8%C3%A5`. This is less legible and transparent than `{"name": "foo bar baz æøå", "script": …}`.

2. If we want to have a command clearing the pinned scripts the most logical endpoint would be `POST /session/{session id}/execute/pin/clear`. It also blocks future extensions of the API, as related commands would need their own top-level endpoint, e.g. `/session/{session id}/execute/specialisedPin`.

Reply:
FWIW I think `POST`ing some data to a URL containing a user-chosen unique id is a reasonable pattern. For deleting it seems reasonable that sending `DELETE` to `/session/{session id}/execute/pin` would work (and that deleting a specific script would be  `DELETE` `/session/{session id}/execute/pin/{name}`).

However I don't want this to be seen as a strong opinion in either direction; the exact details of the HTTP endpoints are not something I particularly care about.

Reply:
I put a proposal in https://github.com/w3c/webdriver/pull/1445#discussion_r334833023 for addressing pinned script deletion. I like the idea of `DELETE /session/{session id}/execute/pin/{name}` deleting a named pinned script and `DELETE /session/{session id}/execute/pin` clearing all. This is similar to how cookies are modelled.

I’m still not convinced using a URI template field name for the POST request is a good idea.

Reply:
Oh, I agree with Andreas here, I think being able to namespace the script name with a colon is important, which can't be done as an endpoint. I'll change it.

Reply:
I think this is all I need to do to fix this. I think I already addressed the things that were overcomplicated the last time I tried this approach... 

https://github.com/w3c/webdriver/pull/1445/commits/4a79a5dbe0ebc079915320005cf3c4b769fb2650